misc/run_length_encoding.c

```diff
@@ -0,0 +1,75 @@
+/**
+ * @file run_length_encoding.c
+ * @author [serturx](https://github.com/serturx/)
+ * @brief Encode a null terminated string using [Run-length encoding](https://en.wikipedia.org/wiki/Run-length_encoding)
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <assert.h>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c
#include <stdio.h>     /// for IO operations
#include <assert.h>    /// for assert
```